### PR TITLE
[RFR] [PATCH] Feature/AddYmlSupport

### PR DESCRIPTION
**Pull request by chrisamoore**
- Replaces https://github.com/strongloop/loopback-swagger/pull/8

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by chrisamoore** [comment]
@bajtos RE: gitter conversation on yaml support.

**Comment by raymondfeng** [comment]
ok to test

**Comment by raymondfeng** [comment]
@chrisamoore Nice. I only see the test yaml file. Where is the test case to verify we only support it?

**Comment by chrisamoore** [comment]
I have to match the JSON spec I think, I just snagged one of the pet store one and it has deltas, I imagine I can copy the test here and use the yml file: https://github.com/strongloop/generator-loopback/blob/master/test/swagger.test.js#L22

**Comment by chrisamoore** [comment]
screen shot 2015-09-12 at 10 47 32 pm 

cc/ @raymondfeng @bajtos

File: swagger/spec-loader.js [hunk 1]
```diff
@@ -62,17 +63,27 @@ function loadSpec(specUrlStr, log, cb) {
       }
     });
   } else {
-    fs.readFile(specUrl.href, 'UTF-8', function (err, body) {
-      if (err) {
-        return cb(err);
-      }
+    if(specUrl.href.match(/.yml/)){
```
Comment (bajtos) [inline swagger/spec-loader.js]: In our coding style, we add spaces after the `if` statement and also between `) {`:

```js
if (specUrl.href.match(/.yml/)) {
```

File: swagger/spec-loader.js [hunk 2]
```diff
@@ -62,17 +63,27 @@ function loadSpec(specUrlStr, log, cb) {
...
       try {
-        var spec = JSON.parse(body);
+        var spec = yaml.safeLoad(fs.readFileSync(specUrl.href, 'utf8'));
         cb(null, spec);
       } catch (err) {
         cb(err);
       }
-    });
+    }else{
```
Comment (bajtos) [inline swagger/spec-loader.js]: Coding style: add spaces around `else` keyword.

```js
} else { 
```

**Comment by bajtos** [comment]
Hi @chrisamoore, thank you for the pull request. The patch looks mostly good, could you please fix the two coding style issues pointed above?

**Comment by bajtos** [comment]
@chrisamoore while you are at fixing coding style, could you please squash all commits into a single one and provide a descriptive commit message following the [50/72 rule](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).

It's ok to say no, in which case I'll make the cleanup myself :)

**Comment by chrisamoore** [comment]
@bajtos Looks like the build failed on sending an email ?

**Comment by chrisamoore** [comment]
ok to test

**Comment by bajtos** [comment]
@slnode test please

**Comment by bajtos** [comment]
Looks like the CI is failing in dependent module due to an unrelated problem. Let's not worry about that in this PR.

**Comment by bajtos** [comment]
Landed and released as `generator-loopback@1.12.0`. Thank you for the contribution!

**Comment by chrisamoore** [comment]
@bajtos Thanks! Docs update?

**Comment by bajtos** [comment]
@chrisamoore I am not sure which docs you would see updated. Could you please contribute the documentation changes too? See https://docs.strongloop.com/display/LB/Contributing+to+LoopBack+documentation

**Comment by bajtos** [comment]
@chrisamoore I am not sure which docs you would like to see updated. Could you please contribute the documentation changes too? See https://docs.strongloop.com/display/LB/Contributing+to+LoopBack+documentation

**Pull request merged**